theano/tensor/nnet/abstract_conv.py

```diff
@@ -366,6 +374,215 @@ def conv2d_grad_wrt_weights(input,
     return gradWeight_op(input, output_grad, filter_shape)
 
 
+def bilinear_kernel_2D(ratio, normalize=True):
+    """ This function gets a ratio
+    and returns the normalized 2D filters that can
+    be applied to any image to upsample it by that ratio
+    using bilinear interpolation.
+
+    :type ratio: integer
+    : param ratio: the ratio by which an image will
+        be upsampled by the returned filters in the 2D space.
+
+    :type normalize: bool
+    : param normalize: indicates whether to normalize the
+    kernel or not. Default is True.
+
+    returns: the 2D kernels that can be applied to
+             any given image to upsample it by the indicated ratio
+             using bilinear interpolation.
+    """
+    assert ratio > 1
+    kern_size = ratio * 2 - 1
+    kern = np.zeros((kern_size, kern_size), dtype=theano.config.floatX)
+    for ii in range(-ratio + 1, ratio):
+        for jj in range(-ratio + 1, ratio):
+            kern[ii + ratio - 1, jj + ratio - 1] = (ratio - abs(ii)) *\
+                                                   (ratio - abs(jj))
+    if normalize:
+        kern /= float(ratio**2)
+    return kern
+
+
+def bilinear_kernel_1D(ratio, normalize=True):
+    """ This function gets a ratio
+    and returns the normalized 1D filters that can
+    be applied to any image to upsample it by that ratio
+    using bilinear interpolation.
+
+    :type ratio: integer
+    : param ratio: the ratio by which an image is upsampled
+        in the 1D space.
+
+    :type normalize: bool
+    : param normalize: indicates whether to normalize the
+    kernel or not. Default is True.
+
+    returns: the 1D kernels that can be applied to
+             any given image to upsample it by the indicated ratio
+             using bilinear interpolation in one dimension.
+    """
+    assert ratio > 1
+    kern_size = ratio * 2 - 1
+    kern = np.zeros((kern_size), dtype=theano.config.floatX)
+    for ii in range(-ratio + 1, ratio):
+        kern[ii + ratio - 1] = ratio - abs(ii)
+    if normalize:
+        kern /= float(ratio)
+    return kern
```

Comment:
Out of curiosity, is it more convenient (or is there any difference) to go for pure Theano or standard list comprehensions, e.g., `[i for i in range(1, ratio+1) + range(ratio-1, 0, -1)]`?